### ISSUE TITLE
MVP: Version control

### ISSUE THREAD

**Issue by gordonleacock** [mlibrary/umrdr#26]
Supports versioning of both deposited items and metadata.

**Comment by gordonleacock** [comment]
What does versioning of metadata mean? Does each version of the item have its own metadata?

**Comment by gordonleacock** [comment]
Should the generic work show a list of versions?

**Comment by grosscol** [comment]
Versioning is done at the level of the ActiveFedora::Base object.

Some relevant code to dig into:
https://github.com/projecthydra-labs/hydra-works/blob/70ec3a9ee7ec11813027f11b8cf386861d862a26/lib/hydra/works/services/add_file_to_file_set.rb#L119-L123

https://github.com/projecthydra/active_fedora/blob/2d60314ae99555cece16f5b3049b6f2add25e67f/lib/active_fedora/versionable.rb

I recall seeing a versioning section of one of the old dive into hydra tutorials when it was first being developed.  No doubt it's out of date now.  It would be useful to have a little tutorial written demonstrating versionable's functionality in the context of the models we are using here.

I've pointed to specific commits so the content of the links won't change over time.

**Comment by njaffer** [comment]
Question: Do we need metadata versioning? File versioning is already there.

**Comment by amyneeser** [comment]
Yes. We need a log of any changes made to anything in Deep Blue Data, including metadata versioning.
